### PR TITLE
Add a helm chart

### LINKED ISSUE #103: Feature Request: Helm chart for cluster-proportional-autoscaler
I would like the Helm Chart to deploy the cluster-proportional-autoscaler.

### PR DESCRIPTION
**Pull request by krmichel**
Fixes #103 

Before this is merged there will need to be a `gh-pages` branch created (probably with nothing but a .gitignore file in it) to house the chart repo's index.yaml.  If you need any help let me know and I can provide some git commands to help with that.  In the event that the `gh-pages` branch is protected here is a PR in another kubernetes-sigs repo that needed to be changed with a reference to a change in test-infra

https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/624

### CODE REVIEW

**Comment by krmichel** [comment]
@MrHohn Here is the PR for the helm chart as promised

File: charts/cluster-proportional-autoscaler/values.yaml
Comment (MrHohn) [inline charts/cluster-proportional-autoscaler/values.yaml:22]: Could you update this to use the multi-arch image - `k8s.gcr.io/cpa/cluster-proportional-autoscaler`

File: charts/cluster-proportional-autoscaler/Chart.yaml
Comment (MrHohn) [inline charts/cluster-proportional-autoscaler/Chart.yaml:3]: Trying to understand - so this is the version being presented in the helm chart? In the future if we need to update the image version, or adjust some configurations in the template file, we will bump this version, is that correct?

**Review by MrHohn — APPROVED** [review]
Thanks @krmichel! I'm not particularly familiar with the helm chart templating, but tried to go over the configuration and RBAC etc. and they looks sane to me.

I have a comment regarding the multi-arch image and will LGTM after that.

/approve

**Comment by MrHohn** [comment]
> Before this is merged there will need to be a gh-pages branch created (probably with nothing but a .gitignore file in it) to house the chart repo's index.yaml.

I can create that once this PR is merged.

File: charts/cluster-proportional-autoscaler/values.yaml
Comment (krmichel) [reply charts/cluster-proportional-autoscaler/values.yaml]: Done

File: charts/cluster-proportional-autoscaler/Chart.yaml
Comment (krmichel) [reply charts/cluster-proportional-autoscaler/Chart.yaml:3]: It sounds like your understanding of the concepts is correct but I will provide additional detail just to help clarify.

Version is the version of the chart itself and will be increased to release new versions of the chart.  Things like changes to the RBAC or adding a network policy, etc.  So, depending on how you want to handle chart releases you will either want chart PRs to increase this number if you want them to release as soon as they are merged to master or if you prefer to delay the release then you would want the PR to not change this version and you would set it on master when ready to release.  As new versions are released they will be listed in the chart repo, which the index for is what will be in the gh-pages branch, allowing people to choose the version of the chart they would like to use.

appVersion is the default version of the proportional autoscaler image that will be used unless image.tag is provided as a value to helm install/upgrade.  This can be seen [here](https://github.com/kubernetes-sigs/cluster-proportional-autoscaler/pull/113/files#diff-b0620dbcddd12f184e07f3e718d25b54793a3420babab54c38087d2d812bce87R36)

**Comment by krmichel** [comment]
> I can create that once this PR is merged.

You can wait until after the merge.  If you do the action to release the chart will fail right after the merge.  However, I think you can then create the branch and go into the actions and tell it to rerun them.

**Comment by krmichel** [comment]
@MrHohn, sorry I think I just accidentally cleared your approval.

**Review by MrHohn — APPROVED** [review]
Thanks for your work @krmichel!
/lgtm
/approve

File: charts/cluster-proportional-autoscaler/Chart.yaml
Comment (MrHohn) [reply charts/cluster-proportional-autoscaler/Chart.yaml:3]: Thank you for the thorough explanation. That is super helpful.

**Comment by MrHohn** [comment]
@krmichel gh-pages branch is created on https://github.com/kubernetes-sigs/cluster-proportional-autoscaler/tree/gh-pages.

**Comment by MrHohn** [comment]
The "Release Helm Charts" job failed initially as you mentioned. I manually rerun it and it is all good now. Let me know if you see any more issue.

**Comment by krmichel** [comment]
@MrHohn I forgot that because the github release was already created the workflow doesn't think it needs to do anything for the chart version.  I think the options are either we delete the release and the tag so the actions sees that it needs to run or I (or someone) make a PR bumping the chart version to 1.0.1 and merge it to make sure it creates the needed file in the gh-pages branch.

**Comment by VioletCranberry** [comment]
Hello guys! 

Sorry for bothering, but is there any update on this workflow? 
The `index.yaml` is missing indeed, so any `repo add` fails as of now: 

```
helm repo add cluster-proportional-autoscaler https://kubernetes-sigs.github.io/cluster-proportional-autoscaler
Error: looks like "https://kubernetes-sigs.github.io/cluster-proportional-autoscaler" is not a valid chart repository or cannot be reached: 
failed to fetch https://kubernetes-sigs.github.io/cluster-proportional-autoscaler/index.yaml : 404 Not Found
```

Thanks in advance! :)

**Comment by MrHohn** [comment]
So I went for the first approach and hit another issue:
```
HEAD is now at a482144 Merge pull request #113 from krmichel/master
[detached HEAD f543c53] Update index.yaml
 1 file changed, 18 insertions(+)
 create mode 100644 index.yaml
Pushing to branch "gh-pages"
remote: error: GH006: Protected branch update failed for refs/heads/gh-pages.        
remote: error: Required status check "cla/linuxfoundation" is expected.        
To https://github.com/kubernetes-sigs/cluster-proportional-autoscaler
 ! [remote rejected] HEAD -> gh-pages (protected branch hook declined)
error: failed to push some refs to 'https://github.com/kubernetes-sigs/cluster-proportional-autoscaler'
Error: exit status 1
```

Looks like we might need something similar to kubernetes-sigs/aws-ebs-csi-driver#624 as you mentioned.

**Comment by krmichel** [comment]
@MrHohn This PR has been merged do you want to delete the release and tag and try again?
https://github.com/kubernetes/test-infra/pull/22996

**Comment by MrHohn** [comment]
@krmichel Thanks for that - I didn't know that config lives in prow :)

Re-run the job and it completes successfully this time.

**Comment by krmichel** [comment]
@VioletCranberry I think it is working now

**Pull request merged**
